### PR TITLE
[Backport][ipa-4-9] freeipa.spec.in: client: depend on libsss_sudo

### PR DESCRIPTION
**Pull request by fcami**
MANUAL BACKPORT of https://github.com/freeipa/freeipa/pull/5176

On 10.10+ releases of Dogtag, the PKI installer will not depend on sudo anymore. This opens the possibility of creating IPA servers without a properly configured sudo.
In fact, even IPA clients should have sudo and libsss_sudo installed in most cases, so: add a weak dependency on libsss_sudo to freeipa-client.

Fixes: https://pagure.io/freeipa/issue/8530

### CODE REVIEW

**Comment by fcami** [comment]
@netoarmando could you please double-check the yaml CI definitions?

**Comment by flo-renaud** [comment]
@fcami in case you're wondering, the tox issue has been reported at https://pagure.io/freeipa/issue/8696

File: ipatests/prci_definitions/nightly_ipa-4-9_latest_selinux.yaml [hunk 1]
```diff
@@ -575,9 +575,22 @@ jobs:
...
+      args:
+        build_url: '{fedora-latest-ipa-4-9/build_url}'
+        selinux_enforcing: True
+        test_suite: test_integration/test_installation.py::TestInstallWithoutSudo
+        template: *ci-ipa-4-9-latest
+        timeout: 4800
+        topology: *master_1repl_1client
+
+  fedora-latest/test_idviews:
+    requires: [fedora-latest/build]
```
Comment (netoarmando) [inline ipatests/prci_definitions/nightly_ipa-4-9_latest_selinux.yaml:592]: It should be: 
```yaml
fedora-latest-ipa-4-9/test_idviews:
    requires: [fedora-latest-ipa-4-9/build]
```

**Comment by abbra** [comment]
@fcami please change the `test_idviews` test reference as requested here: https://github.com/freeipa/freeipa/pull/5533#discussion_r572330020

**Comment by flo-renaud** [comment]
@fcami you probably missed one comment. In `ipatests/prci_definitions/nightly_ipa-4-9_latest_selinux.yaml` line 592:
```
    requires: [fedora-latest/build]
```
should be instead: 
```
    requires: [fedora-latest-ipa-4-9/build]
```

File: ipatests/prci_definitions/nightly_ipa-4-9_latest_selinux.yaml (hunk 1 above)
Comment (fcami) [reply ipatests/prci_definitions/nightly_ipa-4-9_latest_selinux.yaml]: Fixed, thanks.

File: ipatests/prci_definitions/nightly_ipa-4-9_previous.yaml [hunk 2]
```diff
@@ -535,6 +535,19 @@ jobs:
...
         topology: *master_1repl
 
+  fedora-previous-ipa-4-9/test_installation_TestInstallWithoutSudo:
+    requires: [fedora-previous-ipa-4-9/build]
+    priority: 50
+    job:
+      class: RunPytest
+      args:
+        build_url: '{fedora-previous-ipa-4-9/build_url}'
+        update_packages: True
```
Comment (flo-renaud) [inline ipatests/prci_definitions/nightly_ipa-4-9_previous.yaml:545]: update_packages should not be set for 4-9_previous.

Comment (fcami) [reply ipatests/prci_definitions/nightly_ipa-4-9_previous.yaml]: Fixed, thanks for the catch.

**Comment by flo-renaud** [comment]
@fcami thanks, LGTM. ACK for when PRCI completes.

**Comment by rcritten** [comment]
ipa-4-9:

* b590dcef10680b4ea3181ae1caec183e5967562b ipatests: add TestInstallWithoutSudo
* 0c2741af9f353d2fbb21a5768e6433c0e99da0e9 ipatests: tasks: handle uninstalling packages with nodeps
* fe157ca349e3146a53884e90e6e588efb4e97eeb ipa-client-install: output a warning if sudo is not present
* ee0ba2df41cf545b82d3d26e7e7e42447bb0f63e freeipa.spec: client: depend on libsss_sudo and sudo

**Pull request closed without merging**